File: Web/AppServer/AppServer/Models/Signal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace AppServer.Models
{
    /// <summary>
    /// Device signal class
    /// </summary>
    public class Signal : IEntityBase
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public SignalDirection SignalDirection { get; set; }
        public short[] Pattern { get; set; }

        public int DeviceID { get; set; }
    }
}

File: Web/appserver/src/main/java/com/appserver/models/Signal.java
package com.appserver.models;

/**
 * Device signal class
 */
public class Signal extends EntityBase
{
	private int ID;
	private int DeviceID;
	
	private String Name;
	private SignalDirection signalDirection;
	private long[] Pattern;
	
	private String Icon;

    public int getID() { return ID; }
	public void setID(int id) { ID = id; }
	
	//This block requires refactoring, DTO has to be added to store DeviceID
    public int getDeviceID() { return DeviceID; }
	public void setDeviceID(int id) { DeviceID = id; }

    public String getName() { return Name; }
	public void setName(String name) { Name = name; }
	
	public SignalDirection getSignalDirection() {	return signalDirection;	}
	public void setSignalDirection(SignalDirection signalDirection) {	this.signalDirection = signalDirection; }
	
	public long[] getPattern() { return Pattern;}
	public void setPattern(long[] pattern) { Pattern = pattern;}

	public String getIcon() { return Icon;}
	
	public Signal(int id)
	{
		ID = id;
	}	
	public Signal(int id, int deviceid)
	{
		ID = id;
		DeviceID = deviceid;
	}	
}
